src/zap/zapimage.cpp

```diff
@@ -2022,20 +2088,64 @@ ZapImage::CompileStatus ZapImage::TryCompileMethodWorker(CORINFO_METHOD_HANDLE h
     }
 #endif
 
+    // Do we have a profile entry for this method?
+    //
     if (methodProfilingDataFlags != 0)
     {
         // Report the profiling data flags for layout of the EE datastructures
         m_pPreloader->SetMethodProfilingFlags(handle, methodProfilingDataFlags);
 
-        // Only proceed with compilation if the code is hot
+        // Hot methods can be marked to be excluded from the AOT native image.
+        // A Jitted method executes faster than a ReadyToRun compiled method.
+        //
+        if ((methodProfilingDataFlags & (1 << ExcludeHotMethodCode)) != 0)
+        {
+            // returning COMPILE_EXCLUDED excludes this method from the AOT native image
+            return COMPILE_EXCLUDED;
+        }
+
+        // Cold methods can be marked to be excluded from the AOT native image.
+        // We can reduced the size of the AOT native image by selectively
+        // excluding the code for some of the cold methods.
+        //
+        if ((methodProfilingDataFlags & (1 << ExcludeColdMethodCode)) != 0)
+        {
+            // returning COMPILE_EXCLUDED excludes this method from the AOT native image
+            return COMPILE_EXCLUDED;
+        }
```

Comment:
If you returned different status for hot and cold exclusions then the profileDataHashTable could potentially be mined after the fact to display the breakdown of method dispositions.